Lib/concurrent/futures/process.py

```diff
@@ -808,6 +812,8 @@ def submit(self, fn, /, *args, **kwargs):
 
             if self._safe_to_dynamically_spawn_children:
                 self._adjust_process_count()
+                with self._idle_worker_lock:
+                    self._idle_worker_number -= 1
```

Comment:
This is decrementing `_idle_worker_number` just after it was incremented by `_adjust_process_count`, perhaps we can simplify this by adding an optional argument to `_adjust_process_count`, e.g. `self._adjust_process_count(increment_idle_workers=False)`

Reply:
We can. As I said, I think explicitly decrementing the worker number at where a task is submitted is easier to follow. `_adjust_process_count` may or may not increment the idle workers, depending on whether the max worker number is reached. If we combine the option to may or may not decrement the counter, that might be harder to follow. `increment_idle_workers` is not a good idea because it does not really mean anything. `decrement_idle_workers` is acceptable, but still a bit twisted.

The current logic is:
1. `_adjust_process_count()` only adjusts process count, and change `_idle_worker_number` accordingly. It does nothing else.
2. When a new task is submitted, `_idle_worker_number` is decremented
3. When a task is finished, `_idle_worker_number` is incremented

I think this is much easier to follow than "figure out whether `_adjust_process_count()` should decrement the counter", because why would it? The function just `_adjust_process_count`.